src/app/clusters/mode-base-server/mode-base-server.cpp

```diff
@@ -0,0 +1,445 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <app-common/zap-generated/attributes/Accessors.h>
+#include <app/InteractionModelEngine.h>
+#include <app/clusters/mode-base-server/mode-base-server.h>
+#include <app/clusters/on-off-server/on-off-server.h>
+#include <app/reporting/reporting.h>
+#include <app/util/attribute-storage.h>
+#include <platform/DiagnosticDataProvider.h>
+
+// using namespace std;
+using namespace chip;
+using namespace chip::app;
+using namespace chip::app::Clusters;
+using chip::Protocols::InteractionModel::Status;
+using BootReasonType = GeneralDiagnostics::BootReasonEnum;
+
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace ModeBase {
+
+std::map<uint32_t, Instance *> Instance::ModeBaseAliasesInstanceMap;
+
+bool Instance::HasFeature(Feature feature) const
+{
+    return (mFeature & to_underlying(feature)) != 0;
+}
+
+bool Instance::isAliasCluster() const
+{
+    for (unsigned int AliasedCluster : AliasedClusters)
+    {
+        if (mClusterId == AliasedCluster)
+        {
+            return true;
+        }
+    }
+    return false;
+}
+
+CHIP_ERROR Instance::Init()
+{
+    // Check that the cluster ID given is a valid mode base alias cluster ID.
+    if (!isAliasCluster())
+    {
+        ChipLogError(Zcl, "ModeBase: The cluster with ID %lu is not a mode base alias.", long(mClusterId));
+        return CHIP_ERROR_INVALID_ARGUMENT; // todo is this the correct error?
```

Comment:
Yes, but it's often unexpected to have init errors (it blows-up). Suggest a VerifyOrDie() instead since this error should arise once in the lifetime, and you immediately fix it.

Reply:
Is it the responsibility of the init function to kill the application in this case? Or is it the caller's responsibility?

Reply:
I have changed this to a VerifyOrDie [here](https://github.com/project-chip/connectedhomeip/pull/27504/commits/fc0bbbf45fd902340fb9b985af6b999c167ffcc0).